api/src/mcp/tools/assets.ts

```diff
@@ -0,0 +1,69 @@
+import { UnsupportedMediaTypeError } from '@directus/errors';
+import type { TransformationSet } from '@directus/types';
+import { z } from 'zod';
+import { AssetsService } from '../../services/assets.js';
+import { FilesService } from '../../services/files.js';
+import { defineTool } from '../define.js';
+import prompts from './prompts/index.js';
+
+const AssetsValidateSchema = z.strictObject({
+	id: z.string(),
+});
+
+const AssetsInputSchema = z.object({
+	id: z.string(),
+});
+
+export const assets = defineTool<z.infer<typeof AssetsValidateSchema>>({
+	name: 'assets',
+	description: prompts.assets,
+	annotations: {
+		title: 'Directus - Assets',
+	},
+	inputSchema: AssetsInputSchema,
+	validateSchema: AssetsValidateSchema,
+	async handler({ args, schema, accountability }) {
+		const serviceOptions = {
+			accountability,
+			schema,
+		};
+
+		const filesService = new FilesService(serviceOptions);
+
+		const file = await filesService.readOne(args.id, { limit: 1 });
+
+		if (!file.type || !['image', 'audio'].some((t) => file.type?.startsWith(t))) {
+			throw new UnsupportedMediaTypeError({ mediaType: file.type ?? 'unknown', where: 'asset tool' });
+		}
+
+		let transformation: TransformationSet | undefined = undefined;
+
+		// ensure image dimensions are within allowable LLM limits
+		if (file.type.startsWith('image') && file.width && file.height && (file.width > 1200 || file.height > 1200)) {
+			transformation = {
+				transformationParams: {
+					transforms:
+						file.width > file.height
+							? [['resize', { width: 800, fit: 'contain' }]]
+							: [['resize', { height: 800, fit: 'contain' }]],
+				},
+			};
```

Comment:
Not sure if we can do anything about it but these transformations will fail if the `ASSETS_TRANSFORM_MAX_OPERATIONS=0` is set